sdk/Lusid.Sdk.Tests/Utilities/PollyApiRetryHandlerTests.cs

```diff
@@ -0,0 +1,409 @@
+            );
+
+            Assert.That(exception.ErrorContent, Is.EqualTo("{\"some\": \"JsonResponseHere\"}"));
+            Assert.That(exception.ErrorCode, Is.EqualTo(expectedStatusCode));
+            Assert.That(retryCount, Is.EqualTo(0));
+        }
+
+        [Test]
+        public void CallGetPortfoliosApi_WhenHttpStatusIs200AndRetryConditionIsNotSatisfied_NoPollyRetryIsTriggered()
+        {
+            // It should do nothing when response code is 200
+            const int expectedStatusCode = 200;
+            // Add the next response returned by api
+            AddMockHttpResponseToQueue(_httpListener, expectedStatusCode, responseContent: "{\"some\": \"JsonResponseHere\"}");
+            var retryCount = 0;
+            RetryConfiguration.RetryPolicy =
+                Policy.Wrap(
+                    PollyApiRetryHandler.DefaultFallbackPolicy,
+                    Policy
+                        .HandleResult<IRestResponse>(PollyApiRetryHandler.GetInternalExceptionRetryCondition)
+                        .Retry(retryCount: 3,
+                            onRetry: (result, i) => retryCount++)
+                );
+
+            // TODO: This will still fail at deserialization, which currently is not handled and returns null.
+            // Will be done under a different PR. Response string will also be changed.
+            var sdkResponse = _apiFactory.Api<IPortfoliosApi>().GetPortfolio("any", "any");
+
+            Assert.That(retryCount, Is.EqualTo(0));
+        }
+
+        [Test]
+        public void CallGetPortfoliosApi_WhenApiResponseStatusCodeSatisfiesRetryCriteria_ExceedsPollyRetriesAndThrows()
+        {
+            const int returnedStatusCode = 502; // Or any other code that satisfies the policy retry criteria
+            const int expectedNumberOfRetries = 2;
+            const string returnedErrorMessage = "Some error response";
+            for (var i = 0; i < expectedNumberOfRetries + 1; i++)
+                // Every response fails
+                AddMockHttpResponseToQueue(_httpListener, returnedStatusCode, returnedErrorMessage);
+            var retryCount = 0;
+
+            RetryConfiguration.RetryPolicy = Policy.Wrap(
+                PollyApiRetryHandler.DefaultFallbackPolicy,
+                Policy
+                    .HandleResult<IRestResponse>(response => response.StatusCode == (HttpStatusCode) returnedStatusCode)
+                    .Retry(expectedNumberOfRetries, onRetry: (result, i) => retryCount++));
+
+            // Calling GetPortfolio or any other API triggers the flow that triggers polly
+            var exception = Assert.Throws<ApiException>(
+                () => _apiFactory.Api<IPortfoliosApi>().GetPortfolio("any", "any")
+            );
+
+            Assert.That(retryCount, Is.EqualTo(expectedNumberOfRetries));
+            // Todo: In the future 0 error codes with throw an error after retries exceeded
```

Comment:
What does this mean in the context of this test?

Reply:
Means that the response won't just be null, it will actually throw whenever we implement DefaultExceptionFactory